test/Microsoft.AspNet.Mvc.ViewFeatures.Test/ViewFeatures/ViewExecutorTest.cs

```diff
@@ -112,6 +114,59 @@ public class ViewExecutorTest
             Assert.Equal("abcd", Encoding.UTF8.GetString(memoryStream.ToArray()));
         }
 
+        private static IServiceProvider GetServiceProvider()
+        {
+            var httpContext = new HttpContextAccessor() { HttpContext = new DefaultHttpContext() };
+            var serviceCollection = new ServiceCollection();
+            serviceCollection.AddInstance<IModelMetadataProvider>(new EmptyModelMetadataProvider());
+            var tempDataProvider = new SessionStateTempDataProvider();
+            serviceCollection.AddInstance<ITempDataDictionary>(new TempDataDictionary(httpContext, tempDataProvider));
+
+            return serviceCollection.BuildServiceProvider();
+        }
+
+        [Fact]
+        public async Task ExecuteAsync_ViewResultAllowNull()
+        {
+            var tempDataNull = false;
+            var viewDataNull = false;
+            // Arrange
+            var view = CreateView(async (v) =>
+            {
+                tempDataNull = v.TempData == null;
+                viewDataNull = v.ViewData == null;
+
+                await v.Writer.WriteAsync("abcd");
+            });
+            var context = new DefaultHttpContext();
+
+            var memoryStream = new MemoryStream();
+            context.Response.Body = memoryStream;
+
+            var actionContext = new ActionContext(
+                context,
+                new RouteData(),
+                new ActionDescriptor());
+
+
+            context.RequestServices = GetServiceProvider();
+            var viewExecutor = CreateViewExecutor();
+
+            // Act
+            await viewExecutor.ExecuteAsync(
+                actionContext,
+                view,
+                null,
+                null,
+                contentType: null,
+                statusCode: 200);
+
+            // Assert
+            Assert.Equal(200, context.Response.StatusCode);
+            Assert.False(viewDataNull);
+            Assert.False(tempDataNull);
```

Comment:
Please add an assertion showing the delegate calling `WriteAsync()` was actually invoked. Might be worth another Boolean variable.